test/modules/acbs/acbs-authentication.service.test.ts

```diff
@@ -0,0 +1,242 @@
+            await expect(getTokenPromise)
+                .rejects.toHaveProperty('innerError', undefined);
+        });
+
+        it('throws an AcbsAuthenticationFailedException if the IdP returns an empty id_token', async () => {
+            mockSuccessfulCreateSessionRequest();
+            mockSuccessfulGetTokenForSessionRequestReturning({ id_token: '' });
+
+            const getTokenPromise = service.getIdToken();
+
+            await expect(getTokenPromise)
+                .rejects.toBeInstanceOf(AcbsAuthenticationFailedException);
+            await expect(getTokenPromise)
+                .rejects.toThrowError('ID token was not returned by the IdP.');
+            await expect(getTokenPromise)
+                .rejects.toHaveProperty('innerError', undefined);
+        });
+
+        it('throws an AcbsAuthenticationFailedException if the IdP returns an undefined id_token', async () => {
+            mockSuccessfulCreateSessionRequest();
+            mockSuccessfulGetTokenForSessionRequestReturning({ id_token: undefined });
+
+            const getTokenPromise = service.getIdToken();
+
+            await expect(getTokenPromise)
+                .rejects.toBeInstanceOf(AcbsAuthenticationFailedException);
+            await expect(getTokenPromise)
+                .rejects.toThrowError('ID token was not returned by the IdP.');
+            await expect(getTokenPromise)
+                .rejects.toHaveProperty('innerError', undefined);
+        });
+
+        it('throws an AcbsAuthenticationFailedException if the IdP returns a null id_token', async () => {
+            mockSuccessfulCreateSessionRequest();
+            mockSuccessfulGetTokenForSessionRequestReturning({ id_token: null });
+
+            const getTokenPromise = service.getIdToken();
+
+            await expect(getTokenPromise)
+                .rejects.toBeInstanceOf(AcbsAuthenticationFailedException);
+            await expect(getTokenPromise)
+                .rejects.toThrowError('ID token was not returned by the IdP.');
+            await expect(getTokenPromise)
+                .rejects.toHaveProperty('innerError', undefined);
+        });
+    });
+
+    function mockSuccessfulCreateSessionRequest(): void {
+        mockSuccessfulCreateSessionRequestReturningCookies([cookie1, sessionIdCookie, cookie2]);
+    }
+
+    function mockSuccessfulCreateSessionRequestReturningCookies(cookies: string[]): void {
+        const headers = new AxiosHeaders();
+        headers['set-cookie'] = cookies;
+
+        when(httpService.post)
+            .calledWith(...expectedPostSessionsArguments)
+            .mockReturnValueOnce(of({
+                data: '',
+                status: 200,
+                statusText: 'OK',
+                config: undefined,
+                headers: headers,
+            }));
+    }
+
+    function mockSuccessfulGetTokenForSessionRequest(): void {
+        mockSuccessfulGetTokenForSessionRequestReturning({ id_token: idToken });
```

Comment:
Do we need this function if all it does is call another function?

Reply:
I think it's worth it for readability - if we just want the request to be successful then we use this function, and if we need to control the response data then we use the other function.

I considered just using 1 function with an optional argument instead, but you then have to choose if you keep `Returning` in the function name or not, and I think
```javascript
mockSuccessfulTokenForSessionRequest({ id_token: 'some token' })
```
is a bit less clear than keeping `Returning` because it's not clear what the argument is for, and 
```javascript
mockSuccessfulTokenForSessionRequestReturning();
```
is a bit less clear than dropping `Returning` because the function name feels unfinished